### PR TITLE
[MERGE] Merge branch 1.7

### PR DESCRIPTION
**Pull request by aaudiber**
Conflicts resolved in last two commits

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 1]
```diff
@@ -1014,6 +1015,23 @@ public void syncPersistDirectory(InodeDirectory dir, JournalContext journalConte
             MkdirsOptions mkdirsOptions = MkdirsOptions.defaults().setCreateParent(false)
                 .setOwner(dir.getOwner()).setGroup(dir.getGroup()).setMode(new Mode(dir.getMode()));
             ufs.mkdirs(ufsUri, mkdirsOptions);
+            if (!ufs.mkdirs(ufsUri, mkdirsOptions)) {
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: I think this should replace the line above.

Comment (aaudiber) [reply core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: good catch, thanks

**Comment by aaudiber** [comment]
@gpang It looks like these tests are going over the time limit by only 10-20%. Can we safely increase the tolerance on these failing tests so that they don't flake like this?

**Comment by aaudiber** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
@aaudiber I think the test was failing because you were calling 2 mkdirs, instead of 1. It should be fine after your fix.

**Comment by aaudiber** [comment]
@gpang ahh, that makes sense

**Pull request merged**